packages/validation-manager/src/ValidationManager.ts

```diff
@@ -237,7 +237,18 @@ export class ValidationManager implements IValidationManager {
       addresses: [],
       hash: ''
     }
-    const stateOverrideForEip7702 = await this.getAuthorizationsStateOverride(userOp.authorizationList ?? [])
+    const authorizationList = getAuthorizationList(userOp)
+    if (authorizationList.length > 0) {
+      // relevant only for RIP-7562...
+      requireCond(authorizationList.length === 1, 'Only one authorization is supported', ValidationErrors.InvalidFields)
+
+      const currentChainId = BigNumber.from((this.entryPoint.provider as any)._network.chainId)
+      const authChainId = BigNumber.from(authorizationList[0].chainId)
```

Comment:
You converted it into an array, so own up to it - iterate over the list and check all of it. It may become an actual array at any point in time and this will become a bug.